ncm-dpmlfc/src/main/perl/dpmlfc/RuleBasedEditor.pm

```diff
@@ -0,0 +1,817 @@
+line formatting based on the line format.
+
+Arguments :
+    fh : a FileEditor object
+    config_param: parameter to update
+    config_value : parameter value (can be empty)
+    line_fmt : line format (see LINE_FORMAT_xxx constants)
+    multiple : if true, multiple lines with the same keyword can exist (D: false)
+
+=cut
+
+sub _updateConfigLine {
+  my $function_name = "_updateConfigLine";
+  my ($self, $fh, $config_param, $config_value, $line_fmt, $multiple) = @_;
+
+  unless ( $fh ) {
+    $self->error("$function_name: 'fh' argument missing (internal error)");
+    return 1;
+  }
+  unless ( $config_param ) {
+    $self->error("$function_name: 'config_param' argument missing (internal error)");
+    return 1;
+  }
+  unless ( defined($config_value) ) {
+    $self->error("$function_name: 'config_value' argument missing (internal error)");
+    return 1;
+  }
+  unless ( defined($line_fmt) ) {
+    $self->error("$function_name: 'line_fmt' argument missing (internal error)");
+    return 1;
+  }
+  unless ( defined($multiple) ) {
+    $multiple = 0;
+  }
+
+  my $config_param_pattern;
+  my $new_line = $self->_formatConfigLine($config_param,$config_value,$line_fmt);
+
+  # Build a pattern to look for.
+  if ( $multiple ) {
+    $self->debug(2,"$function_name: 'multiple' flag enabled");
+    $config_param_pattern = $self->_buildLinePattern($config_param,$line_fmt,$config_value);    
+  } else {
+    $config_param_pattern = $self->_buildLinePattern($config_param,$line_fmt);
+    if ( ($line_fmt == LINE_FORMAT_XRDCFG) && $config_value ) {
+      $config_param_pattern .= "\\s+";    # If the value is defined in these formats, impose a whitespace at the end
+    }
+  }
+
+  # Update the matching configuration lines
+  if ( $new_line ) {
+    my $comment = "";
+    if ( ($line_fmt == LINE_FORMAT_PARAM) || ($line_fmt == LINE_FORMAT_ENVVAR) ) {
+      $comment = $LINE_QUATTOR_COMMENT;
+    }
+    $self->debug(1,"$function_name: checking expected configuration line ($new_line) with pattern >>>".$config_param_pattern."<<<");
+    $fh->add_or_replace_lines(qr/^\s*$config_param_pattern/,
+                              qr/^\s*$new_line$/,
+                              $new_line.$comment."\n",
+                              ENDING_OF_FILE,
+                             );      
+  }
+}
+
+
+=pod
+
+=item _parse_rules
+
+Apply configuration rules. This method is the real workhorse of the rule-based editor.
+
+Arguments :
+    fh : a FileEditor object
+    config_rules: config rules corresponding to the file to build
+    config_options: configuration parameters used to build actual configuration
+    options: a hash setting options to modify the behaviour of this function
+
+Supported entries for options hash:
+    always_rules_only: if true, apply only rules with ALWAYS condition (D: false)
+    remove_if_undef: if true, remove maatching configuration line is rule condition is not met (D: false)
+
+=cut
+
+sub _parse_rules {
```

Comment:
the rule parser should only parse rules and return bunch of boolean / values, like `$condition, `$negate`,... 
mixing the actual configuration changes makes this hard to read (and a very long function)